cranelift-codegen/src/isa/x86/enc_tables.rs

```diff
@@ -954,3 +954,64 @@ fn convert_extractlane(
         }
     }
 }
+
+/// Because floats exist in XMM registers, we can keep them there when executing a CLIF
+/// insertlane instruction
+fn convert_insertlane(
+    inst: ir::Inst,
+    func: &mut ir::Function,
+    _cfg: &mut ControlFlowGraph,
+    _isa: &dyn TargetIsa,
+) {
+    let mut pos = FuncCursor::new(func).at_inst(inst);
+    pos.use_srcloc(inst);
+
+    if let ir::InstructionData::InsertLane {
+        opcode: ir::Opcode::Insertlane,
+        args: [vector, replacement],
+        lane,
+    } = pos.func.dfg[inst]
+    {
+        // remove the original extractlane instruction
+        pos.remove_inst();
+
+        // extract the original result value and use it (e.g. with_result) as the final result value of our replaced sequences
+        let original_results = pos.func.dfg.detach_results(inst);
+        let original_result = original_results.get(0, &pos.func.dfg.value_lists).unwrap();
+
+        let value_type = pos.func.dfg.value_type(vector);
+        if value_type.lane_type().is_float() {
+            // floats are already in XMM registers and can stay there
+            match value_type {
+                F32X4 => {
+                    assert!(lane > 0 && lane <= 3);
```

Comment:
Is there anything in the code base that disallow lanes out of that range by throwing an error? (i'd expect so in the verifier) If not, this is something we need (in another PR if you prefer).

Reply:
I know that at least `wasmparser` is doing those checks here: https://github.com/yurydelendik/wasmparser.rs/blob/master/src/binary_reader.rs#L1256-L1265.

Reply:
It'd be nice to open a follow-up to make sure that we actually throw an error in the verifier's code, if that happens; we may have other producers than cranelift-wasm.

Reply:
Opened https://github.com/CraneStation/cranelift/issues/1016.